sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.institutions.controller.ts

```diff
@@ -0,0 +1,95 @@
+import { Controller, Get, Param, ParseIntPipe } from "@nestjs/common";
+import BaseController from "../BaseController";
+import {
+  RestrictionInstitutionDetailAPIOutDTO,
+  RestrictionInstitutionSummaryAPIOutDTO,
+} from "./models/restriction.dto";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+} from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import { ApiNotFoundResponse, ApiTags } from "@nestjs/swagger";
+import { RestrictionControllerService } from "./restriction.controller.service";
+import { getUserFullName } from "../../utilities";
+
+/**
+ * Controller for Institution Restrictions.
+ * This consists of all Rest APIs for Institution restrictions.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@IsBCPublicInstitution()
+@HasStudentDataAccess("studentId")
+@Controller("restriction")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-restriction`)
+export class RestrictionInstitutionsController extends BaseController {
+  constructor(
+    private readonly restrictionControllerService: RestrictionControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get restrictions for a student filtered with restriction notification type other than `No effect`.
+   * @param studentId id of the student to retrieve restrictions.
+   * @returns Student restrictions.
+   */
+  @Get("student/:studentId")
+  async getStudentRestrictions(
+    @Param("studentId", ParseIntPipe) studentId: number,
+  ): Promise<RestrictionInstitutionSummaryAPIOutDTO[]> {
+    const studentRestrictions =
+      await this.restrictionControllerService.getStudentRestrictions(
+        studentId,
+        {
+          filterNoEffectRestrictions: true,
+        },
+      );
+    return studentRestrictions?.map((studentRestriction) => ({
+      restrictionId: studentRestriction.id,
+      restrictionType: studentRestriction.restriction.restrictionType,
+      restrictionCategory: studentRestriction.restriction.restrictionCategory,
+      restrictionCode: studentRestriction.restriction.restrictionCode,
+      description: studentRestriction.restriction.description,
+      createdAt: studentRestriction.createdAt,
+      isActive: studentRestriction.isActive,
+    }));
+  }
+
+  /**
+   * Get the details for view student restriction.
+   * @param studentId id of the student.
+   * @param studentRestrictionId id of the student restriction.
+   * @returns Student restriction detail view.
+   */
+  @Get("student/:studentId/studentRestriction/:studentRestrictionId")
```

Comment:
I missed it in the previous reviews but the `studentRestriction` should be either `student-restriction` or just `restriction`. I noticed on Swagger that we used in the past  `studentRestriction` and `institutionRestriction` (that is why I will not ask to change) but this is not the API endpoint convention.
@ann-aot @dheepak-aot @andrepestana-aot @guru-aot FYI

Reply:
oh... we might have missed it in the past, but as you suggest it should be `student-restriction` or `restriction`